openedx-unsupported/configuration
# SE-1186 Enhances the analytics_single.yml playbook

Modifies the `analytics_single.yml` playbook to be able to use it to set up single-VM production analytics servers.

**JIRA tickets**: [OSPR-3695](https://openedx.atlassian.net/browse/OSPR-3695)

**Sandbox URL**:

* Insights: https://se-1186-analytics-test.sandbox.opencraft.hosting
* Analytics API: https://se-1186-analytics-test.sandbox.opencraft.hosting:4330

**Merge deadline**: None

**Testing instructions**:

We haven't set up DNS for this server, so you'll need an `/etc/hosts` entry to map the IP address to the hostname:
```
54.38.245.235 se-1186-analytics-test.sandbox.opencraft.hosting
```

1. Ensure that Insights is running and accessible over SSL.
   (There's only a self-signed certificate, and authentication won't work, since OAuth has not been set up to a running edxapp instance.)
1. Ensure that the analytics API is running and accessible over SSL.
   (Also using a self-signed certificate)
   
**Author Notes & Concerns**

1. We had issues getting the `analytics_pipeline` role to run in our sandbox environment because it's not running a local mysql instance.  The role tries to [create mysql users](https://github.com/open-craft/configuration/blob/jill/analytics_server/playbooks/roles/analytics_pipeline/tasks/main.yml#L223-L231), but isn't configured to connect to mysql on a different host, so throws: 
   ```
   FAILED! => {"changed": false, "msg": "unable to connect to database, check login_user and login_******** are correct or /root/.my.cnf has the credentials. Exception message: (2002, \"Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)\")"}
   ```
   Not sure what uses this role?  Our use case only requires `insights` and the `analytics_api` to be running, so we disabled the pipeline with: `SANDBOX_ENABLE_ANALYTICS_PIPELINE: false`

**Reviewers**
- [ ] @gr4scale

Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [x] Are you adding any new default values that need to be overridden when this change goes live? No
